source/reactor/inc/reactor/EventLoop.h

```diff
@@ -0,0 +1,73 @@
+#ifndef Reactor_EventLoop_H
+#define Reactor_EventLoop_H
+
+namespace u {
+namespace reactor {
+class EventHandlers;
+
+/**
+ * The event loop is a programming construct that waits for and dispatches events in a program.
+ *
+ * It works by making a request to some "event provider" (which generally blocks the request
+ * until an event has arrived), and then it calls the relevant event handler (i.e.: dispatches
+ * the event).
+ *
+ * The event provider used by the event loop follows the file interface, which can be selected
+ * or 'polled' (the Unix system call, not actual polling). The event loop operates
+ * asynchronously with the event originator.
+ *
+ * When the event loop forms the central control flow construct of a program, as it often does,
+ * it may be termed the main loop or main event loop. This title is appropriate because such an
+ * event loop is at the highest level of control within the program.
+ */
+class EventLoop {
+public:
+    virtual ~EventLoop() = default;
+
+    /**
+     * @brief Registers a set of event handlers for the given file descriptor.
+     *
+     * Event handler for the event that occurred will be called back when the file descriptor is
+     * ready for a read/write O operation, when an error occurs or when given timeout expires.
+     *
+     * @param fd File descriptor. Handle that identifies a resource managed within an OS:
+     * network connections, open files, timers, synchronization objects, etc.
+     * @param timeout Time to wait in milliseconds (-1 to wait indefinitely) for a read/write
+     * event to occur.
+     * @param handlers Set of event handlers: class with the methods to be called back when an
+     * event occurs.
+     * @return True on success and false otherwise.
+     */
+    virtual bool registerHandlers(int fd, long timeout, const EventHandlers &handlers) = 0;

```

Comment:
If we have timeouts, I think we should use an expressive type for them. `std::chrono::milliseconds` seems extremely appropriate.

Reply:
Actually you stole my comment... But past experience shows that the following approach is even better:
```suggestion
    template<class Duration>
    virtual bool registerHandlers(int fd, Duration timeout, const EventHandlers &handlers) = 0;
```
So the caller can have the timeout in any `std::chrono` she likes.
(C++20 gives us _restriction_ (predicate) on the Duration type, but C++11 does not)